### PR TITLE
Polish documentation for RTD.

### LINKED ISSUE #39: Online docs are outdated
The [GenericSetup PyPI page](https://pypi.python.org/pypi/Products.GenericSetup) points to documentation at http://packages.python.org/Products.GenericSetup/ and https://pythonhosted.org/Products.GenericSetup/. The first url redirects to the second one.

What we see there, is documentation for version 1.7.0. Does anyone know how to update this?

Alternatively, we could rearrange things and move to readthedocs.

### PR DESCRIPTION
**Pull request by icemac**
See the result on RTD: http://productsgenericsetup.readthedocs.io/en/fix-39/
In contrast http://productsgenericsetup.readthedocs.io/en/latest/ shows the current `master` branch.

The changes in `docs/conf.py` are adopted from zopefoundation/z3c.form#69

Fixes #39

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -55,12 +55,19 @@ def _package_doc(name):
         'six',
         'Zope2 >= 4.0a6',
         'five.localsitemanager',
+        'AccessControl >= 4.0a4',
```
Comment (tseaver) [inline setup.py]: This pin doesn't seem related to the PR.

File: docs/conf.py [hunk 2]
```diff
@@ -177,3 +188,49 @@
 
 # If false, no module index is generated.
 #latex_use_modindex = True
+
+# -- Options for manual page output ---------------------------------------
+
+# One entry per manual page. List of tuples
+# (source start file, name, description, authors, manual section).
+man_pages = [
+    ('index', 'z3cform', 'z3c.form Documentation',
```
Comment (mauritsvanrees) [inline docs/conf.py]: Here and below, there are a couple of `z3c.form` mentions that should be `Products.GenericSetup`.

Comment (icemac) [reply docs/conf.py]: Sorry for copy-pasting without reading it through again.

File: setup.py (hunk 1 above)
Comment (icemac) [reply setup.py]: Sadly it is required, without this pin `tox -edocs` ends in

`ContextualVersionConflict: (AccessControl 3.0.13 (/.../Products.GenericSetup/.tox/docs/lib/python2.7/site-packages), Requirement.parse('AccessControl>=4.0a4'), set(['Products.ZCatalog']))`

Interestingly this does not happen when running the tests.

But there is a way around this: Using `pip_pre = True`. Are you fine with it?

**Comment by icemac** [comment]
@tseaver @mauritsvanrees Are you okay with my changes?
